ee/clickhouse/migrations/0028_remove_dollar_from_materialized_columns.py

```diff
@@ -0,0 +1,48 @@
+import structlog
+from clickhouse_driver.errors import ServerException
+from infi.clickhouse_orm import migrations
+
+from ee.clickhouse.materialized_columns.columns import get_materialized_columns, materialized_column_name
+from posthog.client import sync_execute
+from posthog.constants import GROUP_TYPES_LIMIT
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_REPLICATION
+
+logger = structlog.get_logger(__name__)
+
+
+def rename_column(table, current_name, new_name):
+    if CLICKHOUSE_REPLICATION and table == "events":
```

Comment:
Does this migration work properly for sharding?

CLICKHOUSE_REPLICATION is always going to be true now so are you sure that the sharding migration has run by this point? If not, we're making a whole set of instances unupgradable.